atacac/_utils.py

```diff
@@ -126,3 +162,12 @@ def load_asset(file_path):
             return yaml.safe_load(f)
     except IOError:
         raise Error(f"Failed to read content of '{file_path}'")
+
+
+def sanitize_filename(filename):
+    """
+    Sanitize filename so it contains only alphanumeric characters, dot,
+    underscore or dash. All other characters are replaced with single
+    underscore.
+    """
+    return re.sub(r'[^a-zA-Z0-9.-]+', '_', filename)
```

Comment:
> All other characters are replaced with single underscore.

It is **false**, in some cases they are just excluded (not replaced with `_`).

```
Python 3.7.7 (default, Mar 13 2020, 21:39:43) 
[GCC 9.2.1 20190827 (Red Hat 9.2.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import re
>>> filename = "a/b.c_d-e"
>>> re.sub(r'[^a-zA-Z0-9.-]+', '_', filename)
'a_b.c_d-e'
>>> filename = "a/_b"
>>> re.sub(r'[^a-zA-Z0-9.-]+', '_', filename)
'a_b'
>>> filename = "a#_b"
>>> re.sub(r'[^a-zA-Z0-9.-]+', '_', filename)
'a_b'
>>> 
```

Reply:
It is expected, consecutive underscores are collapsed into one eg. `a_%$#@_b` should be transformed to `a_b`. I updated doc.